server/app/services/PageNumberBasedPaginationSpec.java

```diff
@@ -0,0 +1,35 @@
+package services;
+
+import java.util.Optional;
+
+/**
+ * Specifies pagination behavior for a query with page number-based pagination. With page-number
+ * based pagination, the current page of results is resolved by multiplying the page size by the
+ * current page - 1 to derive an offset for the the underlying query.
+ */
+public class PageNumberBasedPaginationSpec {
+
+  public static PageNumberBasedPaginationSpec MAX_PAGE_SIZE_SPEC =
+      new PageNumberBasedPaginationSpec(Integer.MAX_VALUE, 1);
+
+  private final int pageSize;
+  private final Optional<Integer> currentPage;
+
+  public PageNumberBasedPaginationSpec(int pageSize) {
+    this.pageSize = pageSize;
+    this.currentPage = Optional.empty();
```

Comment:
Is this constructor still necessary after removing the other class from being a child?

This will cause getCurrentPage() to throw an exception with no way to avoid it.

Reply:
done and removed the optional